server/modules/category/service/categoryImportJob.js

```diff
@@ -17,231 +18,248 @@ const CategoryImportJobParams = require('./categoryImportJobParams')
 
 class CategoryImportJob extends Job {
 
-  constructor (params) {
-    super(CategoryImportJob.type, params)
+  constructor (params, type = CategoryImportJob.type) {
+    super(type, params)
 
     this.itemsByCodes = {} // cache of category items by ancestor codes
+    this.summary = null
+    this.category = null
+    this.levelIndexDeepest = 0 // used to remove unused levels
   }
 
   async execute () {
-    // levels
-    let category = await this._importLevels()
-
-    // item extra def
-    category = await this._importItemExtraDef(category)
-
-    // items
-    await this._readItems(category)
-
-    if (this.hasErrors()) {
-      // errors found
+    // 1. initialize summary (get it from params by default)
+    this.summary = await this.getOrCreateSummary()
+
+    this.logDebug('summary', this.summary)
+
+    //skip import if summary is not specified
+    if (!this.summary)
+      return
+
+    // 2. fetch or create category
+    this.category = await this._fetchOrCreateCategory()
+
+    // 3. import levels
+    await this._importLevels()
+    // 4. import item extra def
+    await this._importItemExtraDef()
+    // 5. read items from csv file
+    await this._readItems()
+
+    if (this.total === 0) {
+      // error: empty file
+      this._addError(ValidatorErrorKeys.categoryImport.emptyFile)
+    } else if (this.hasErrors()) {
+      // errors found in csv rows
       this.logDebug(`${Object.keys(this.errors).length} errors found`)
       await this.setStatusFailed()
     } else {
       // no errors found
-      await this.insertItems()
+      // 6. delete unused levels
+      this.category = await CategoryManager.deleteLevelsFromIndex(this.user, this.surveyId, this.category, this.levelIndexDeepest, this.tx)
```

Comment:
why do you need CategoryManager.deleteLevelsFromIndex? would the code be shorter if you just delete all levels and items before importing them? why do you replace  the old ones and then remove the unused levels?